src/analysis/sensitivity.jl

```diff
@@ -32,6 +32,38 @@ function ks_statistic(ks::ApproximateKSTest)::Float64
     return sqrt(n) * ks.δ
 end
 
+"""
+    _get_factor_spec(model_spec::DataFrame, factors::Vector{String})
+
+Get model spec for specified factors.
+"""
+function _get_factor_spec(model_spec::DataFrame, factors::Vector{Symbol})
+    factors_to_assess = model_spec.fieldname .∈ [factors]
+    foi_attributes = Symbol[:fieldname, :ptype, :lower_bound, :upper_bound]
+    return model_spec[factors_to_assess, foi_attributes]
+end
+
+"""
+    _set_cat_S(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+
+Get number of binnings for categorical variables.
+"""
+function _set_cat_S(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+    max_bounds = maximum(foi_spec[foi_cat, :upper_bound] .- foi_spec[foi_cat, :lower_bound])
+    return round(Int64, max(S, max_bounds))
+end
```

Comment:
Function name is a bit cryptic - I'm not sure I understand what a `cat_S` is.

It also does not set anything?

What about `_category_bins()`?

Reply:
By `cat_S` I meant categorical S, but I'll change to `_category_bins`